platform/src/main/java/common/org/tron/common/utils/MarketComparator.java

```diff
@@ -0,0 +1,124 @@
+package org.tron.common.utils;
+
+import java.math.BigInteger;
+
+public class MarketComparator {
+
+  public static final int TOKEN_ID_LENGTH = Long.toString(Long.MAX_VALUE).getBytes().length; // 19
+
+
+  public static int comparePriceKey(byte[] o1, byte[] o2) {
+    //compare pair
+    byte[] pair1 = new byte[TOKEN_ID_LENGTH * 2];
+    byte[] pair2 = new byte[TOKEN_ID_LENGTH * 2];
+
+    System.arraycopy(o1, 0, pair1, 0, TOKEN_ID_LENGTH * 2);
+    System.arraycopy(o2, 0, pair2, 0, TOKEN_ID_LENGTH * 2);
+
+    int pairResult = compareUnsigned(pair1, pair2);
+    if (pairResult != 0) {
+      return pairResult;
+    }
+
+    //compare price
+    byte[] getSellTokenQuantity1 = new byte[8];
+    byte[] getBuyTokenQuantity1 = new byte[8];
+
+    byte[] getSellTokenQuantity2 = new byte[8];
+    byte[] getBuyTokenQuantity2 = new byte[8];
+
+    int longByteNum = 8;
+
+    System.arraycopy(o1, TOKEN_ID_LENGTH + TOKEN_ID_LENGTH,
+        getSellTokenQuantity1, 0, longByteNum);
+    System.arraycopy(o1, TOKEN_ID_LENGTH + TOKEN_ID_LENGTH + longByteNum,
+        getBuyTokenQuantity1, 0, longByteNum);
+
+    System.arraycopy(o2, TOKEN_ID_LENGTH + TOKEN_ID_LENGTH,
+        getSellTokenQuantity2, 0, longByteNum);
+    System.arraycopy(o2, TOKEN_ID_LENGTH + TOKEN_ID_LENGTH + longByteNum,
+        getBuyTokenQuantity2, 0, longByteNum);
+
+    long sellTokenQuantity1 = toLong(getSellTokenQuantity1);
+    long buyTokenQuantity1 = toLong(getBuyTokenQuantity1);
+    long sellTokenQuantity2 = toLong(getSellTokenQuantity2);
+    long buyTokenQuantity2 = toLong(getBuyTokenQuantity2);
+
+    if ((sellTokenQuantity1 == 0 || buyTokenQuantity1 == 0)
+        && (sellTokenQuantity2 == 0 || buyTokenQuantity2 == 0)) {
+      return 0;
+    }
+
+    if (sellTokenQuantity1 == 0 || buyTokenQuantity1 == 0) {
+      return -1;
+    }
+
+    if (sellTokenQuantity2 == 0 || buyTokenQuantity2 == 0) {
+      return 1;
+    }
+
+    return comparePrice(sellTokenQuantity1, buyTokenQuantity1,
+        sellTokenQuantity2, buyTokenQuantity2);
+
+  }
+
+  /**
+   * Note: the params should be the same token pair, or you should change the order.
+   * All the quantity should be bigger than 0.
+   * */
+  public static int comparePrice(long price1SellQuantity, long price1BuyQuantity,
```

Comment:
`comparePrice` is already exist in org.tron.core.capsule.utils.MarketUtils. Should it be extracted and reused ?